velox/common/file/File.cpp

```diff
@@ -254,16 +261,38 @@ LocalWriteFile::~LocalWriteFile() {
 
 void LocalWriteFile::append(std::string_view data) {
   VELOX_CHECK(!closed_, "file is closed");
-  const uint64_t bytes_written = fwrite(data.data(), 1, data.size(), file_);
+  errno = 0;
+  const uint64_t bytesWritten = fwrite(data.data(), 1, data.size(), file_);
   VELOX_CHECK_EQ(
-      bytes_written,
+      bytesWritten,
       data.size(),
       "fwrite failure in LocalWriteFile::append, {} vs {}: {}",
-      bytes_written,
+      bytesWritten,
       data.size(),
       folly::errnoStr(errno));
 }
 
+void LocalWriteFile::append(std::unique_ptr<folly::IOBuf> data) {
+  VELOX_CHECK(!closed_, "file is closed");
+  uint64_t bytesWritten{0};
+  errno = 0;
+  for (auto rangeIter = data->begin(); rangeIter != data->end(); ++rangeIter) {
+    bytesWritten += fwrite(rangeIter->data(), 1, rangeIter->size(), file_);
+    if (errno != 0) {
```

Comment:
Shall we just check on the bytes to write and the number of written bytes? Thanks!